overlord/ifacestate/ifacestate.go

```diff
@@ -321,9 +322,47 @@ func Disconnect(st *state.State, conn *interfaces.Connection) (*state.TaskSet, e
 	return disconnectTasks(st, conn, disconnectOpts{})
 }
 
+// Forget returs a set of tasks for disconnecting and forgetting an interface.
+// If the interface is already disconnected, it will be removed from the state
+// (forgotten).
+func Forget(st *state.State, repo *interfaces.Repository, connRef *interfaces.ConnRef) (*state.TaskSet, error) {
+	if err := snapstate.CheckChangeConflictMany(st, []string{connRef.PlugRef.Snap, connRef.SlotRef.Snap}, ""); err != nil {
+		return nil, err
+	}
+
+	var disconnectTask *state.Task
+	var ts *state.TaskSet
+	conn, err := repo.Connection(connRef)
+	if err == nil {
+		opts := disconnectOpts{}
```

Comment:
shouldn't we set Forget here? and not use the edge, seems the code is solving the same problem two different ways

Reply:
Good point, thanks.